src/helpers.ts

```diff
@@ -3,14 +3,19 @@ import type { ErrorLike } from './types';
 export function checkValidArgs(keyValuePairs: unknown[], callback: unknown) {
   if (
     !Array.isArray(keyValuePairs) ||
-    keyValuePairs.length === 0 ||
-    !Array.isArray(keyValuePairs[0])
+    (keyValuePairs.length > 0 && !Array.isArray(keyValuePairs[0]))
   ) {
     throw new Error(
       '[AsyncStorage] Expected array of key-value pairs as first argument to multiSet'
     );
   }
 
+  if (keyValuePairs.length === 0) {
+    console.warn(
+      '[AsyncStorage] Expected non-empty array of key-value pairs as first argument to multiSet'
+    );
+  }
```

Comment:
let's make sure it works with a zero-length array

Reply:
Sure, will test

Reply:
Tested on Egors PR before the fix & it works
```diff
diff --git a/src/_local_modules/app/wallet/migrations/migrate-market-history-storage.js b/src/_local_modules/app/wallet/migrations/migrate-market-history-storage.js
index f5cc3d97b7..7e4f8d3ec8 100644
--- a/src/_local_modules/app/wallet/migrations/migrate-market-history-storage.js
+++ b/src/_local_modules/app/wallet/migrations/migrate-market-history-storage.js
@@ -60,10 +60,7 @@ export default async function migrateMarketHistoryStorage({
     return acc
   }, {})

-  if (Object.keys(updates).length > 0) {
-    await namespaced.batchSet(updates)
-  }
-  if (toDeleteKeys.length > 0) {
-    await legacyStorage.remove(toDeleteKeys)
-  }
+  await namespaced.batchSet(updates)
+  await legacyStorage.remove(toDeleteKeys)
+  console.log('migration success')
 }
```